pennylane/tape/tape.py

```diff
@@ -1036,12 +1037,70 @@ def get_depth(self):
 
         >>> tape.get_depth()
         4
+
         """
+
+        warnings.warn(
+            "``tape.get_depth`` will be deprecated after v0.16 "
+            "Please use the more general ``tape.specs`` instead.",
+            UserWarning,
+        )
+
         if self._depth is None:
             self._depth = self.graph.get_depth()
 
         return self._depth
 
+    @property
+    def specs(self):
+        """Resource requirements of a quantum circuit.
+
+        Returns:
+            dict[str, Union[defaultdict,int]]: how many times constituent operations are applied
```

Comment:
```suggestion
            dict[str, Union[defaultdict,int]]: dictionaries that contain information on how many times constituent operations are applied
```
(maybe)

Reply:
It also seems more than just the number of times ops are applied:
```
{'by_size': defaultdict(int, {1: 3, 2: 1}),
 'by_name': defaultdict(int, {'RX': 1, 'RY': 2, 'CNOT': 1}),
 'total_operations': 4,
 'total_observables': 1,
 'num_tape_wires': 2,
 'depth': 3,
 'num_trainable_params': 3}
```